examples/request_raw.rs

```diff
@@ -0,0 +1,191 @@
+//! Simple example that demonstrates how to use raw API Server requests.
+//! Raw requests are the equivalent of `kubectl get --raw`, and enable
+//! users to query for objects or data that is not available through the
+//! clientsets (i.e. Api<K>, whether primitive or derived from a CRD).
+//! The example builds a tool similar to `kubectl top nodes`.
+use k8s_openapi::{api::core::v1::Node, apimachinery::pkg::api::resource::Quantity};
+use kube::{api::ListParams, Api, ResourceExt};
+use serde::Deserialize;
+
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let client = kube::Client::try_default().await?;
+
+    let api: Api<Node> = Api::all(client.clone());
+    let nodes = api.list(&ListParams::default()).await?;
+
+    let node_names = nodes.items.iter().map(|n| n.name_any()).collect();
+    let mut table = Table::new(node_names);
+
+    for node in nodes.items {
+        // Query node stats by issuing a request to the admin endpoint.
+        // See https://kubernetes.io/docs/reference/instrumentation/node-metrics/
+        let url = format!("/api/v1/nodes/{}/proxy/stats/summary", node.name_any());
+        let req = http::Request::get(url).body(Default::default())?;
+
+        // Deserialize JSON response as a JSON value. Alternatively, a type that
+        // implements `Deserialize` can be used.
+        let resp = client.request::<serde_json::Value>(req).await?;
+
+        // Our JSON value is an object so we can treat it like a dictionary.
+        let summary = resp
+            .get("node")
+            .expect("node summary should exist in kubelet's admin endpoint");
+
+        // The base JSON representation includes a lot of metrics, including
+        // container metrics. Use a `NodeMetric` type to deserialize only the
+        // values we care about.
+        let node_metric = serde_json::from_value::<NodeMetric>(summary.to_owned())?;
+
+        // Get the current allocatable values for the node we are looking at and
+        // save in a table we will use to print the results.
+        let alloc = node.status.unwrap_or_default().allocatable.unwrap_or_default();
+        table.push(node_metric, alloc);
+    }
+
+    table.print();
+
+    Ok(())
+}
+
+/// A stat table made up of rows. Holds a value for the longest name to properly
+/// right pad columns.
+#[derive(Debug, Default)]
+struct Table {
+    rows: Vec<(NodeMetric, NodeAlloc)>,
```

Comment:
this type is slightly confusing to me. you have `rows` which is a vec of nodes and node properties.

does it make sense to create a `NodeSummary` struct here instead and pass a `Vec<NodeSummary>` around? the `Table` type only justifies itself because of the `max_name_width` which can be computed at the end in print

..this way `print_table` can take be a free-standing fn that takes a `Vec<NodeSummary>` and you don't need a separate `push` fn (since you are just making a vec). you also don't need to pre-iterate over the the nodes to collect the node_names because you are putting the node_name into the summary.

Reply:
Thanks, all of the suggestions are great and make sense. It provides a much needed simplification :) I went ahead and make most of the changes. I also debated on `print` taking the max width (since we already have access to the node name in the first loop so we can compute a max) but in retrospect it felt more intuitive to me to do it all in the print function.

Let me know how it looks. At first I did this more as a prototype to answer the discussions so if you spot any other bits I'm happy to make changes 👀

Reply:
Looks great. 👍 

> in retrospect it felt more intuitive to me to do it all in the print function.

yeh, it's also more separation-of-concerns friendly which is good for examples. printing does printing, main does parsing, then delegate.